src/interface/outputs.h

```diff
@@ -0,0 +1,228 @@
+#pragma once
+
+#include "core/bit_board.h"
+#include "core/time_manager.h"
+#include "search/searcher.h"
+#include "version.h"
+
+#include <fmt/color.h>
+
+#include <memory>
+
+#ifdef _WIN32
+#include <io.h>
+#define isatty _isatty
+#define fileno _fileno
+#else
+#include <unistd.h>
+#endif
+
+namespace {
+
+static const bool s_stdoutIsTty = isatty(fileno(stdout)) != 0;
+static inline bool s_isPrettyPrintEnabled = s_stdoutIsTty;
+
+inline void printSearchInfoUci(std::shared_ptr<search::Searcher> searcher, const BitBoard& board, Score score, uint8_t currentDepth, uint64_t nodes, uint64_t tbHits)
+{
+    const auto timeDiff = TimeManager::timeElapsedMs().count();
+
+    const auto adjustedScore = searcher->approxDtzScore(board, score);
+    const uint16_t hashFull = core::TranspositionTable::getHashFull();
+
+    fmt::print("info score {} time {} depth {} seldepth {} nodes {} hashfull {}{}{} pv ",
+        ScorePrint(adjustedScore),
+        timeDiff,
+        currentDepth,
+        searcher->getSelDepth(),
+        nodes,
+        hashFull,
+        NpsPrint(nodes, timeDiff),
+        TbHitPrint(tbHits));
+
+    fmt::println("{}", fmt::join(searcher->getPvTable(), " "));
+}
+
+inline void printSearchInfoPretty(std::shared_ptr<search::Searcher> searcher, const BitBoard& board, Score score, uint8_t currentDepth, uint64_t nodes, uint64_t tbHits)
+{
+    using namespace fmt;
+    using namespace std::chrono_literals;
+
+    const auto timeDiff = TimeManager::timeElapsedMs();
+    const auto knps = nodes / (timeDiff.count() + 1);
+
+    const auto adjustedScoreCp = searcher->approxDtzScore(board, score) / 100.f;
+    const auto mateScore = scoreMateDistance(score);
+    const auto scoreColor = score < 0 ? fg(color::red) : fg(color::lawn_green);
+
+    const auto hashFull = core::TranspositionTable::getHashFull() / 10.0;
+    const auto selDepth = searcher->getSelDepth();
+
+    std::string scoreBuffer;
```

Comment:
@clarashepherd just a note: for these small strings we should not worry about heap allocation. SSO will be used :)